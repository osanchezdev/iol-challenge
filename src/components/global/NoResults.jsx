import React from 'react';
import {Link} from 'react-router-dom';
import {Badge} from 'react-bootstrap';

import '../../styles/components/global/NoResults.scss';

const NoResults = () => {
  return (
    <div className="no-results--wrapper">
      <div className="no-results--emoji">
        <span>&#128533;</span>
      </div>
      <h5 className="no-result--text">No results, try searching something else</h5>
      <div className="no-results--link">
        <Link
          to={{
            pathname: '/',
            search: '',
          }}>
          <Badge pill variant="light">
            Reset filters
          </Badge>
        </Link>
      </div>
    </div>
  );
};

export default NoResults;
