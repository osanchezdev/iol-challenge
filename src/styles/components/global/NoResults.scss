.no-results--wrapper {
  width: 100%;
  text-align: center;
  & .no-results--emoji {
    font-size: 70px;
  }
}
