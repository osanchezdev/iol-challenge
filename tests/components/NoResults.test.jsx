import * as React from 'react';
import {shallow} from 'enzyme';
import NoResults from '../../src/components/global/NoResults';

describe('Tests in <Noresults />', () => {
  test('<Noresults /> must be rendered correctly', () => {
    const wrapper = shallow(<NoResults />);

    expect(wrapper).toMatchSnapshot();
  });
});
