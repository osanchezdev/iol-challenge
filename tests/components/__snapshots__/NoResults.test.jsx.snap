// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on Loader Loader must be rendered correctly 1`] = `
<div
  className="loader--wrapper"
>
  <Spinner
    animation="border"
    variant="light"
  />
</div>
`;

exports[`Test on NoResults NoResults must be rendered correctly 1`] = `
<ForwardRef(MotionComponent)
  animate={
    Object {
      "opacity": 1,
      "scale": 1,
    }
  }
  className="no-results--wrapper"
  initial={
    Object {
      "opacity": 0,
      "scale": 0,
    }
  }
  transition={
    Object {
      "damping": 20,
      "stiffness": 260,
      "type": "spring",
    }
  }
>
  <div
    className="no-results--emoji"
  >
    <span>
      ðŸ˜•
    </span>
  </div>
  <h5
    className="no-result--text"
  >
    No results, try searching something else
  </h5>
  <div
    className="no-results--link"
  >
    <Link
      to={
        Object {
          "pathname": "/",
          "search": "",
        }
      }
    >
      <Badge
        pill={true}
        variant="light"
      >
        Reset filters
      </Badge>
    </Link>
  </div>
</ForwardRef(MotionComponent)>
`;

exports[`Tests in <Noresults /> <Noresults /> must be rendered correctly 1`] = `
<ForwardRef(MotionComponent)
  animate={
    Object {
      "opacity": 1,
      "scale": 1,
    }
  }
  className="no-results--wrapper"
  initial={
    Object {
      "opacity": 0,
      "scale": 0,
    }
  }
  transition={
    Object {
      "damping": 20,
      "stiffness": 260,
      "type": "spring",
    }
  }
>
  <div
    className="no-results--emoji"
  >
    <span>
      ðŸ˜•
    </span>
  </div>
  <h5
    className="no-result--text"
  >
    No results, try searching something else
  </h5>
  <div
    className="no-results--link"
  >
    <Link
      to={
        Object {
          "pathname": "/",
          "search": "",
        }
      }
    >
      <Badge
        pill={true}
        variant="light"
      >
        Reset filters
      </Badge>
    </Link>
  </div>
</ForwardRef(MotionComponent)>
`;

exports[`Tests in <Noresults /> NoResults must be rendered correctly 1`] = `
<ForwardRef(MotionComponent)
  animate={
    Object {
      "opacity": 1,
      "scale": 1,
    }
  }
  className="no-results--wrapper"
  initial={
    Object {
      "opacity": 0,
      "scale": 0,
    }
  }
  transition={
    Object {
      "damping": 20,
      "stiffness": 260,
      "type": "spring",
    }
  }
>
  <div
    className="no-results--emoji"
  >
    <span>
      ðŸ˜•
    </span>
  </div>
  <h5
    className="no-result--text"
  >
    No results, try searching something else
  </h5>
  <div
    className="no-results--link"
  >
    <Link
      to={
        Object {
          "pathname": "/",
          "search": "",
        }
      }
    >
      <Badge
        pill={true}
        variant="light"
      >
        Reset filters
      </Badge>
    </Link>
  </div>
</ForwardRef(MotionComponent)>
`;
